pkg/tuple/tuple.go

```diff
@@ -95,7 +95,7 @@ func ToObjectRelationString(object, relation string) string {
 
 // GetUserTypeFromUser returns the type of user (userset or user).
 func GetUserTypeFromUser(user string) UserType {
-	if IsObjectRelation(user) || user == "*" {
+	if IsObjectRelation(user) || IsWildcard(user) {
```

Comment:
This is nuanced since it changes the `user_type` value written to the database. User values such as `user:*` are also treated as `userset` values now (where they weren't previously). However, this doesn't negatively impact Check behavior as far as I can tell. The string literal `user:*` is still treated as a string literal in Checks with 1.0 models but now `user:*` is treated as a typed wildcard for 1.1 models.

@craigpastro @miparnisari please be careful to review this and make sure I'm not overlooking something here.

Reply:
There is a test in `storage/test/tuples.go` called `reading userset tuples that exists succeeds`. Can you change this test to include a `user:*` tuple?

Reply:
There is a unit test elsewhere in the codebase titled `"reading userset tuples that exists succeeds"`. Can you edit it to show what happens when there is a tuple with `user:*`?

Reply:
Without a migration this is the only way to handle it. Makes sense to me.

Reply:
@miparnisari I've updated the storage tests to reflect the new behavior for typed wildcards. Namely, `storage.Write(...@user:*)` writes as a 'userset' row and `ReadUserTuple` and `ReadUsersetTuples` will return tuples involving typed wildcards.

How does that look?